installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -85,8 +95,23 @@ public static ActionResult InstallStateTool(Session session, out string stateToo
             }
 
             string windowsZip = "windows-amd64.zip";
-            string zipURL = stateURL + info.version + "/" + windowsZip;
             string zipPath = Path.Combine(tempDir, windowsZip);
+            if (File.Exists(zipPath))
+            {
+                try
+                {
+                    File.Delete(zipPath);
+                }
+                catch (Exception e)
+                {
+                    string msg = string.Format("Could not delete existing temporary State Tool zip file at: {0}, encountered exception: {1}", zipPath, e.ToString());
+                    session.Log(msg);
+                    ActiveState.RollbarHelper.Report(msg);
+                    return ActionResult.Failure;
+                }
+            }
+
+            string zipURL = stateURL + info.version + "/" + windowsZip;
```

Comment:
This is needlessly error prone. Just generate a random dir name for the tempDir (you can use a timestamp).